packages/vite/src/node/plugins/asset.ts

```diff
@@ -428,3 +433,28 @@ export async function urlToBuiltUrl(
     true,
   )
 }
+
+// Inspired by https://github.com/iconify/iconify/blob/main/packages/utils/src/svg/url.ts
+function svgToDataURL(content: Buffer): string {
+  const stringContent = content.toString()
+  // If the SVG contains some text, any transformation is unsafe, and given that double quotes would then
+  // need to be escaped, the gain to use a data URI would be ridiculous if not negative
+  if (stringContent.includes('<text')) {
+    return `data:image/svg+xml;base64,${content.toString('base64')}`
+  } else {
+    return (
+      'data:image/svg+xml,' +
+      stringContent
+        .trim()
+        .replaceAll('"', "'")
+        .replaceAll('%', '%25')
+        .replaceAll('#', '%23')
+        .replaceAll('<', '%3c')
+        .replaceAll('>', '%3e')
```

Comment:
Thanks a lot for this feature, for the sake of which I'm switching to v5-beta without waiting for a release!

But these two characters don't need to be encoded. [Here is an example](https://codepen.io/firefoxic/pen/NWMPoOd?editors=0100) where both characters work in all three major browser engines (and it seems like a long time ago).

I'm not sure if a pull request is needed.

Reply:
Interesting! Effectively the wikipedia page says that this is not reserved: https://en.wikipedia.org/wiki/Percent-encoding
A bit more precision on wether it was at one point not supported and if is it like ie11 ages or three years ago will help decide on this change.
Anyway a PR for discussing this change is welcome, you can tag me in

Reply:
I guess it's required for the cases like `url(data:image/svg+xml,<svg viewBox='0 0 32 16' xmlns='http://www.w3.org/2000/svg'><path d='M0 7v2h25v4l7-5-7-5v4z'/></svg>)` (note that the value is not quoted).

Reply:
We should already be auto-quoting SVGs here, so I think it should be fine in most cases: https://github.com/vitejs/vite/blob/788145790e7652d715575d7a0bc33a825b49d566/packages/vite/src/node/plugins/css.ts#L1522-L1536

If the user passes the SVG string to a manually constructed `url()` through some JS framework, they would also need to double quote it to safely pass it today 🤔 I guess it's a tradeoff.

Reply:
> Anyway a PR for discussing this change is welcome, you can tag me in

@ArnaudBarre, done: #14815 🙂

Reply:
From  Vue2. If it's written like something similar to this, it's going to be a problem.

```Vue
export default class Overview extends Vue {
      @Prop({ type: Object, default: () => ({}) }) readonly config!: CardConfig;

      get iconStyle() {
        return {
            width: this.iconSize,
            height: this.defaultHeight + 'px',
            background: 'url(' + this.config.image + ')',
            backgroundSize: 'contain',
            backgroundRepeat: 'no-repeat',
            backgroundPositionY: 'center'
        };
    }
}
```